src/sources/csv.rs

```diff
@@ -0,0 +1,264 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs::File, path::PathBuf};
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{
+	config::FeatureFlags,
+	user::{StringOrBytes, User},
+	zitadel::SourceDiff,
+	Config, FeatureFlag,
+};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+	/// Optional set of features
+	feature_flags: FeatureFlags,
```

Comment:
The `feature_flag` source is present in the CSV source and in the LDAP source. Those sources need it. The UKT source doesn't need it, so it's not there.

Reply:
`LdapSourceConfig` doesn't have `feature_flags` at the moment. Neither should `CsvSource`

Reply:
Wait, I was looking somewhere else, nevermind the last comment

Reply:
Both `CsvSource` and `LdapSource` have the same structure except the base config (`feature_flags` and `idp_id`). Which means, that implementors of `Source` are using the same common config values to, apparently, do the same thing. So I'm wondering why, and can we factor it out? It is a consequence of us using zitadel's `User` inside `Source` definition. We could create an intermediate user type, but I'm also fine using it because some fields may come from the sources, and not set for all users from a config. I'd suggest to construct data inside `Source` implementors from the data *really* comes from those sources. And only at later stage, independently where data came from, fill those `idp_ids` and `email_should_be_verified`

Reply:
Done.